packages/now-routing-utils/src/superstatic.ts

```diff
@@ -201,13 +201,25 @@ function replaceSegments(
           .filter(val => typeof val === 'string') as string[]
       );
 
-      pathname = safelyCompile(pathname, indexes);
-      hash = hash ? safelyCompile(hash, indexes) : null;
+      try {
+        // Attempt compiling normally with path-to-regexp first and fall back
+        // to safely compiling to handle edge cases if path-to-regexp compile
+        // fails
+        pathname = compile(pathname, { validate: false })(indexes);
+        hash = hash ? compile(hash, { validate: false })(indexes) : null;
```

Comment:
Should the `pathname` and `hash` have separate try/catch instead so you can catch which one failed?

Reply:
Good point, moved the direct compiling attempt into `safelyCompile` behind a param so it can be used for non-header related values. Header values only use `safelyCompile` since they aren't expected to be URLs per-se and can have more edge cases so the directly compiling attempt is skipped for them